src/mpi/datatype/typerep/src/typerep_yaksa_create.c

```diff
@@ -7,6 +7,33 @@
 #include "typerep_internal.h"
 #include "yaksa.h"
 
+/* yaksa may disagree with mpich on some internal fields. To ensure consistency,
+ * we update these fields here by querying yaksa */
+#define YAKSA_UPDATE_NEWTYPE(newtype) \
+    do { \
+        yaksa_type_t dt = (yaksa_type_t) (intptr_t) newtype->typerep.handle; \
+        uintptr_t num_contig_blocks; \
+        int cnt = 2; \
+        rc = yaksa_iov_len(cnt, dt, &num_contig_blocks); \
+        MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa"); \
+        if (num_contig_blocks == 1) { \
+            newtype->is_contig = 1; \
+            newtype->typerep.num_contig_blocks = 1; \
+        } else { \
+            newtype->is_contig = 0; \
+            newtype->typerep.num_contig_blocks = (MPI_Aint) num_contig_blocks / cnt; \
+        } \
+        rc = yaksa_type_get_size(dt, &newtype->size); \
+        MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa"); \
+        rc = yaksa_type_get_extent(dt, &newtype->lb, &newtype->extent); \
+        MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa"); \
+        newtype->ub = newtype->lb + newtype->extent; \
+        MPI_Aint true_extent; \
+        rc = yaksa_type_get_true_extent(dt, &newtype->true_lb, &true_extent); \
+        MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa"); \
+        newtype->true_ub = newtype->true_lb + true_extent; \
+    } while (0)
+
```

Comment:
Why is this a macro?  It's not performance critical, and it doesn't require any state from the calling function.  In general, we have been moving away from macros in MPICH towards either functions or static inlines.

Reply:
I'll make it into a function